source/util.cpp

```diff
@@ -36,25 +36,21 @@ namespace binder {
 
 
 /// Split string using given separator
-vector<string> split(string const &buffer, string const &separator)
-{
-	string line;
-	vector<string> lines;
-
-	for( uint i = 0; i < buffer.size(); ++i ) {
-		if( buffer.compare(i, separator.size(), separator) ) line.push_back(buffer[i]);
-		else {
-			lines.push_back(line);
-			line.resize(0);
-		}
-	}
-
-	if( line.size() ) lines.push_back(line);
-
-	return lines;
+std::vector<std::string> split(const std::string &s, const std::string &delimiter) {
+    size_t pos_start = 0, pos_end, delim_len = delimiter.length();
+    std::string token;
+    std::vector<std::string> res;
+
+    while ((pos_end = s.find(delimiter, pos_start)) != std::string::npos) {
+        token = s.substr (pos_start, pos_end - pos_start);
+        pos_start = pos_end + delim_len;
+        res.push_back (token);
+    }
+
+    res.push_back (s.substr (pos_start));
+    return res;
```

Comment:
Could you please say a few words why do you want to rewrite this function? Thanks,